models/performance_profile_persister.go

```diff
@@ -0,0 +1,94 @@
+package models
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/gofrs/uuid"
+	"github.com/layer5io/meshkit/database"
+)
+
+// PerformanceProfilePersister is the persister for persisting
+// performance profiles on the database
+type PerformanceProfilePersister struct {
+	DB *database.Handler
+}
+
+// PerformanceProfilePage represents a page of performance profiles
+type PerformanceProfilePage struct {
+	Page       uint64                `json:"page"`
+	PageSize   uint64                `json:"page_size"`
+	TotalCount int                   `json:"total_count"`
+	Profiles   []*PerformanceProfile `json:"profiles"`
+}
+
+// GetPerformanceProfiles returns all of the performance profiles
+func (ppp *PerformanceProfilePersister) GetPerformanceProfiles(userID, search, order string, page, pageSize uint64) ([]byte, error) {
+	if order == "" {
+		order = "updated_at desc"
+	}
+
+	count := int64(0)
+	profiles := []*PerformanceProfile{}
+
+	query := ppp.DB.
+		Select(`
+		id, name, load_generators, 
+		endpoints, qps, service_mesh, 
+		duration, request_headers, request_cookies, 
+		request_body, content_type, created_at,
+		updated_at, (?) as last_run, (?) as total_results`,
+			ppp.DB.Table("meshery_results").Select("MAX(meshery_results.test_start_time)").Where("performance_profile = performance_profiles.id"),
```

Comment:
@utkarsh-pro The subquery works then?

Reply:
@kumarabd to some extent it does but we lose the type conversion for some reason. That's why I had to change the type to `string` rather than `time.Time`.

But yeah, it finally did :tada:

Reply:
yes a custom type needs to be defined for time.Time as well. Basically a separate valuer/scanner interface